python/pyspark/testing/utils.py

```diff
@@ -138,6 +140,32 @@ def setUpClass(cls):
     def tearDownClass(cls):
         cls.sc.stop()
 
+    def check_error(
+        self,
+        exception: PySparkException,
+        error_class: str,
+        message_parameters: Optional[Dict[str, str]] = None,
+    ):
+        # If exception is generated as a context of assertRaises,
+        # get an actual exception from the context.
+        if isinstance(exception, unittest.case._AssertRaisesContext):
```

Comment:
Is this standard way of doing this by using the class start with `_` prefix classes?

Reply:
Oh... we should handle this in the test case itself, let me address it.